src/content/blog/kai-deep-dive-2024/index.mdx

```diff
@@ -0,0 +1,171 @@
+
+This area of handling cascaded changes is what we refer to as 'Phase 2' and is the next set of development efforts the team is undertaking. Phase 2, inspired by the work of Microsoft in their [CodePlan: Repository-level Coding using LLMs and Planning](https://arxiv.org/abs/2309.12499) paper involves cascading changes throughout a repository of code. We can detect what the changed code "touches" by leveraging the LSP server (similar to what we already use in [analyzer-lsp](https://github.com/konveyor/analyzer-lsp)). Each of those changes is then fed back into the algorithm, over and over, until no more changes are necessary.
+
+![Code Plan ](Kai_CascadingChanges.png)
+
+#### #3 Desire to work with knowledge not in the model
+
+The nature of modernization engagements in organizations is that a large number of the issues faced deal with internal propietary frameworks. It is unlikely that an existing model will have data on these frameworks. We are leveraging the RAG pattern described previously to help mitigate this concern by supplying few-shot prompt examples with extra contextual information of how the organization has solved this problem in the past.
+
+Going beyond this RAG pattern we see future work paths to ease the integration of exposing Konveyor's data into on-premise AI platforms such as [Open Data Hub](https://opendatahub.io/). This integration would support an organization's ability for fine tuning a local model on the data they have collected in Konveyor.
+
+#### #4 Model capabilities are quickly improving
+
+We've built Kai to be model agnostic so organizations may experiment with new models as they are released.
+
+Kai may be configured with coordinates to a LLM provider, allowing for use with public AI providers, internally hosted LLMs, or even local LLMs. We introduced a concept of being able to tweak generated prompts based on the 'family' of LLM being used to help facilitate freedom with exploring various models.
+
+#### #5 Iterate on responses with a LLM to improve the solution
+
+We are building the notion of an Agent into Kai to help improve the quality of a LLM's final solution by working with the LLM to iterate on a smaller scoped solution and then running that solution through several tools to help check the validity and go back to the LLM to address problems found.
+
+![Agent Approach](Agent_Approach.png)
+
+## Next Steps?
+
+The team is working towards integration with Konveyor in Summer 2024, yet plans to remain in the [konveyor-ecosystem](https://github.com/konveyor-ecosystem) for a few more months as the solution is implemented and improved, we expect to pursue being an official Konveyor component later in 2024.
```

Comment:
If we want engagement we should probably lead with the call to action, this makes it seem like I shouldn't engage with it at all for now (which could be want we want, just highlighting it).